[Docs] Add phpdoc

chtipepere opened the pull request:
First PR to get in touch.

pablodip commented on .gitignore:
```diff
@@ -2,3 +2,4 @@ composer.lock
 Guardfile
 phpunit.xml
 /vendor/
+.idea/
```
Personal preferences should not be in the repository.

You can use a global gitignore for this: https://help.github.com/articles/ignoring-files/

pablodip commented on src/Pagerfanta/Pagerfanta.php:
```diff
@@ -105,6 +105,11 @@ public function getNormalizeOutOfRangePages()
         return $this->normalizeOutOfRangePages;
     }
 
+    /**
```
I'm -1 to use PHPDoc unless it helps for something.

For me documenting is helpful to know things you can't guess.

Documenting private functions doesn't help much.
Documenting variable names which are some lines below doesn't help much, overall with good variable names which already indicate what they have.
Documenting "mixed" returns don't help much.

pablodip replied:
Documenting is also helpful for me to indicate the IDE things that can't be guessed.

stof commented on src/Pagerfanta/Pagerfanta.php:
```diff
@@ -134,6 +139,11 @@ public function setMaxPerPage($maxPerPage)
     }
 
+    /**
+     * @param $maxPerPage
```
all your phpdoc is invalid btw. It is missing the type between ``@param`` and the name

stof commented on src/Pagerfanta/Pagerfanta.php:
```diff
@@ -105,6 +105,11 @@ public function getNormalizeOutOfRangePages()
+    /**
+     * @param $value
+     *
+     * @return mixed
```
the return value is always a boolean

stof commented on src/Pagerfanta/View/Template/DefaultTemplate.php:
```diff
@@ -28,11 +28,19 @@ class DefaultTemplate extends Template
+    /**
+     * @param int $page
+     *
+     * @return mixed
```
the return value is always a string

stof commented on src/Pagerfanta/View/Template/DefaultTemplate.php:
```diff
@@ -28,11 +28,19 @@ class DefaultTemplate extends Template
     );
 
+    /**
+     * @return mixed
```
this is a string

stof commented on src/Pagerfanta/View/Template/DefaultTemplate.php:
```diff
@@ -40,6 +48,12 @@ public function page($page)
+     * @param int    $page
+     * @param string $text
+     *
+     * @return mixed
```
always a string

stof commented on src/Pagerfanta/View/Template/Template.php:
```diff
@@ -26,11 +26,17 @@ public function __construct()
     }
 
+    /**
+     * @param array $options
```
this does not add any value compared to the method signature

stof commented on src/Pagerfanta/View/Template/Template.php:
```diff
@@ -26,11 +26,17 @@ public function __construct()
     }
 
+    /**
+     * @param $routeGenerator
```
this phpdoc *could* be useful if it was documenting the type. Given it does not, it is totally useless (and invalid)

chtipepere commented:
First, thanks for your answers, really.
Thanks too for the global gitignore ;)

I've reverted some of the PHPDOC I've added. I also normalized Boolean/bool and integer/int typings in existing PHPDOC.

stof commented on src/Pagerfanta/Pagerfanta.php:
```diff
@@ -60,9 +60,9 @@ public function getAdapter()
+     * @param bool $value
      *
-     * @return self
+     * @return Pagerfanta
```
why changing this ?

chtipepere replied:
Question of habits. But habits can be changed;)

stof commented on src/Pagerfanta/Pagerfanta.php:
```diff
@@ -98,13 +98,19 @@ public function setNormalizeOutOfRangePages($value)
     }
 
+    /**
+     * @param bool $value
```
the input is not always a boolean (which is precisely the goal of this method

chtipepere replied:
Indeed.

stof commented on src/Pagerfanta/View/DefaultView.php:
```diff
@@ -31,11 +33,17 @@ class DefaultView implements ViewInterface
     private $endPage;
 
+    /**
+     * @param TemplateInterface $template
```
this does not add any value at all compared to the signature. It only duplicates knowledge (and with a mistake as the actual type is ``TemplateInterface|null``)

chtipepere replied:
Good point.

stof commented on src/Pagerfanta/View/DefaultView.php:
```diff
@@ -54,6 +62,9 @@ public function render(PagerfantaInterface $pagerfanta, $routeGenerator, array $
     }
 
+    /**
+     * @param PagerfantaInterface $pagerfanta
```
there is no value here either

stof commented on src/Pagerfanta/View/DefaultView.php:
```diff
@@ -238,64 +280,3 @@ public function getName()
-    color: #cf;
-}
-
-*/
```
why removing this ?

chtipepere replied:
Why is there some css rules in this php file?
If it's useful for the implementation, why not put it in the readme? Or just add comments above to explain why it's here?

chtipepere replied:
In fact, it's already in the README.
